### PR TITLE
Improve API doc for ZipDataset

### PR DESCRIPTION
**Pull request by frreiss**
This PR fills out the API documentation for the `ZipDataset` op. I extended the arguments lists to match the operator's definition, and I added a few paragraphs of description, including information about the different code paths by which a `ZipDataset` op might show up in a graph.

### CODE REVIEW

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt [hunk 1]
```diff
@@ -1,5 +1,37 @@
...
+  attr {
+    name: "output_shapes"
+  }
+  attr {
+    name: "N"
+    description: "The length of `input_datasets`"
+  }
   summary: "Creates a dataset that zips together `input_datasets`."
+  description: <<END
+The resulting dataset will produce a stream of elements, where elements `k`
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: Replace with: "The resulting dataset will produce a stream of elements, whose elements are created by zipping corresponding elements from each of the input datasets."

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt [hunk 2]
```diff
@@ -1,5 +1,37 @@
...
+contains the concatenation of the `k`th elements from each of the input
+datasets.
+
+The stream will stop as soon as any one of the input datasets runs out 
+of elements, and noerror will be raised if the other datasets contain
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: "no error"

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt [hunk 3]
```diff
@@ -1,5 +1,37 @@
...
+  description: <<END
+The resulting dataset will produce a stream of elements, where elements `k`
+contains the concatenation of the `k`th elements from each of the input
+datasets.
+
+The stream will stop as soon as any one of the input datasets runs out 
+of elements, and noerror will be raised if the other datasets contain 
+additional data.
+
+`ZipDataset` invokes its input datasets serially, one record at a time.
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: this is an implementation detail

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt [hunk 4]
```diff
@@ -1,5 +1,37 @@
...
+
+Most instances of this type of dataset come from the `zip()` method in
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: I would remove this paragraph.

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt (hunk 1 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: Updated this sentence accordingly in commit 8c5390c

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt (hunk 2 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: Fixed this typo in commit 8c5390c

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt (hunk 3 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: Removed this sentence in commit 8c5390c

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt (hunk 4 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: Removed this paragraph in commit 8c5390c

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt [hunk 5]
```diff
@@ -1,5 +1,29 @@
...
+  attr {
+    name: "output_shapes"
+  }
+  attr {
+    name: "N"
+    description: "The length of `input_datasets`"
+  }
   summary: "Creates a dataset that zips together `input_datasets`."
+  description: <<END
+The resulting dataset will produce a stream of elements, whose elements are
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: replace `stream of elements` with a `dataset` ... let's not introduce redundant synonyms

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt [hunk 6]
```diff
@@ -1,5 +1,29 @@
...
+  attr {
+    name: "N"
+    description: "The length of `input_datasets`"
+  }
   summary: "Creates a dataset that zips together `input_datasets`."
+  description: <<END
+The resulting dataset will produce a stream of elements, whose elements are 
+created by zipping corresponding elements from each of the input datasets.
+
+The stream will stop as soon as any one of the input datasets runs out
```
Comment (jsimsa) [inline tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: The size of the resulting dataset will match the size of the smallest input dataset.

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt (hunk 5 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: Reworded this sentence in commit 771666c

File: tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt (hunk 6 above)
Comment (frreiss) [reply tensorflow/core/api_def/base_api/api_def_ZipDataset.pbtxt]: Reworded this sentence as requested in commit 771666c

**Review by jsimsa — APPROVED** [review]
Thank you!

**Pull request merged**